lib/prop/limiter.rb

```diff
@@ -134,22 +136,28 @@ def handles
 
       private
 
-      def at_threshold?(mark, threshold)
-        mark >= threshold
-      end
-
-      def disabled?
-        !!@disabled
-      end
-
       def prepare(handle, key, params)
-        raise RuntimeError.new("No such handle configured: #{handle.inspect}") unless (handles || {}).key?(handle)
+        raise RuntimeError.new("No such handle configured: #{handle.inspect}") unless (Prop::Limiter.handles || {}).key?(handle)
 
-        defaults  = handles[handle]
+        defaults  = Prop::Limiter.handles[handle]
         options   = Prop::Options.build(:key => key, :params => params, :defaults => defaults)
-        cache_key = Prop::Key.build(:key => key, :handle => handle, :interval => options[:interval])
 
-        [ options, cache_key ]
+        if options[:leaky_bucket]
+          strategy = Prop::LeakyBucketStrategy
+
+          options[:burst_rate] = options.fetch(:burst_rate).to_i
```

Comment:
(Doing that means you don't have to change the return value of this function, since it'd be in `options`)